content_sync/pipelines/definitions/concourse/site-pipeline.yml

```diff
@@ -108,7 +108,7 @@ jobs:
             args:
             - -exc
             - |
-              mkdir ../ocw-hugo-themes/theme
+              mkdir -p ../ocw-hugo-themes/theme
               tar -xvzf ../ocw-hugo-themes/ocw-hugo-themes-*.tgz -C ../ocw-hugo-themes/theme
```

Comment:
I'm wondering though, does this re-running create the possibility of some 'leakyness,' where some files could be in `../ocw-hugo-themes/theme` from an older theme? like for instance if a file is changed the unpacking the `.tgz` archive will overwrite it, but if a file were delete (and not present in the newer theme) then we wouldn't overwrite it with anything

maybe this isn't relevant at all given how the builds are triggered (i.e. there's no possibility of a retry pulling in a newer version of the theme than the original 'try') but I don't think I fully grok how the concourse stuff gets triggered and whatnot.

so anyhow - maybe not anything to worry about, but just something I thought I'd raise

Reply:
You're right to be concerned, but this situation only happens during "retries."  The version of the theme that a given pipeline run is pulling in is set at the beginning of the build.  In the PR I noted above, Matt added `attempts: 3` to each step which means if a step in the pipeline times out or fails for any reason, it will retry it with the same inputs.  This is to cover cases of timeouts looking for workers or API outages, etc.  When a new build is started, all fresh docker containers are built.

Reply:
ok sounds good!